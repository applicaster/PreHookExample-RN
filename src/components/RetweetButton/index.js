import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ActionButton from '../ActionButton';
import { RETWEET_ICON } from '../../icons';

class RetweetButton extends Component {
  render() {
    const { retweetCount } = this.props;

    return (
      <ActionButton
        imageUri={RETWEET_ICON}
        label={retweetCount}
        onPress={() => {}}
      />
    );
  }
}

RetweetButton.propTypes = {
  eventId: PropTypes.string,
  retweetCount: PropTypes.number,
};

RetweetButton.contextTypes = {
  textColor: PropTypes.string,
};

export default RetweetButton;
