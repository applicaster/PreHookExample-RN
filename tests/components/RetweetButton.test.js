import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import RetweetButton from '../../src/components/RetweetButton';

jest.mock('@applicaster/feed-rn-utils', () => 'FeedRNUtils');
jest.mock('react-native-zapp-bridge', () => 'NativeZappBridge');

describe('RetweetButton', () => {
  test('component renders correctly', () => {
    const html = renderer.create(
      <RetweetButton eventId={'1'} retweetCount={1} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
