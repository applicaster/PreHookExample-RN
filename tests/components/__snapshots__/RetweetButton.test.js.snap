// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetweetButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
      "marginRight": 7,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={
      Object {
        "uri": "feed_rn_retweet_button",
      }
    }
    style={
      Array [
        Object {
          "height": 30,
          "marginRight": 1,
          "width": 30,
        },
        Object {
          "tintColor": "#FFFFFF",
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        Object {
          "color": "#FFFFFF99",
        },
      ]
    }
  >
    1
  </Text>
</View>
`;
